src/WebJobs.Script/Extensions/HttpRequestExtensions.cs

```diff
@@ -102,6 +104,38 @@ public static bool IsPlatformInternalRequest(this HttpRequest request, IEnvironm
             return string.Compare(value, "true", StringComparison.OrdinalIgnoreCase) == 0;
         }
 
+        public static bool IsInternalAuthAllowed(this HttpRequest httpRequest)
+        {
+            // First verify that this is an internal request
+            if (!httpRequest.IsAppServiceInternalRequest())
+            {
+                return false;
+            }
+
+            // Next check to see if the specific API is allowed based on whitelist
+            var options = httpRequest.HttpContext.RequestServices.GetService<IOptions<FunctionsHostingConfigOptions>>();
```

Comment:
Have not checked, but if the rest of the codebase is using `IOptionsMonitor` for this, let's do the same here to avoid running through the setup again.

Reply:
All the consumers (but one) I see are taking either an `IOptions<FunctionsHostingConfigOptions>` or a `FunctionsHostingConfigOptions` directly.